### PR TITLE
feat(terraform): add input variable file for greater customization options

### PR DESCRIPTION
**Pull request by chbul**

### CODE REVIEW

File: .github/workflows/terraform.yml [hunk 1]
```diff
@@ -39,6 +39,11 @@ on:
         required: false
         default: ubuntu-latest
 
+      var_file:
+        description: Variable file with regional-specific values
```
Comment (sefornes) [inline .github/workflows/terraform.yml:43]: Wouldn't it be better to specify the description as something a bit more neutral? I mean, since you're referencing a .tfvars file, it could be anything. Not just region specific stuff.

Comment (hknutsen) [reply .github/workflows/terraform.yml:43]: I agree 👍 

How about something like this?

> The path, relative to the working directory, of a variable definitions (`.tfvars`) file containing values for variables declared in the root module.

**Review by hknutsen — CHANGES_REQUESTED** [review]

File: .github/workflows/terraform.yml (hunk 1 above)
Comment (hknutsen) [reply .github/workflows/terraform.yml:43]: ^ This is very similar to the description we already use for input `backend_config`:

> The path, relative to the working directory, of a configuration file containing the remaining arguments for a partial backend configuration.

**Comment by hknutsen** [comment]
@chbul PR title should no longer refer to "region", as this is no longer reflected in the contents of the PR.

File: .github/workflows/terraform.yml [hunk 2]
```diff
@@ -133,12 +139,21 @@ jobs:
...
+        env:
+          TFVARS_FILE: ${{ inputs.var_file }}
         run: |
-          terraform plan -input=false -out="$TF_PLAN_FILE" -detailed-exitcode
+          OPTIONAL_ARGS=()
+          if [[ "$TFVARS_FILE" != "null" ]]; then
+            OPTIONAL_ARGS+=(-var-file="${TFVARS_FILE}")
+          fi
+
+          terraform plan -input=false -out=tfplan -detailed-exitcode "${OPTIONAL_ARGS[@]}"
```
Comment (hknutsen) [inline .github/workflows/terraform.yml:150]: Also, undo this change:

```suggestion
          terraform plan -input=false -out="$TF_PLAN_FILE" -detailed-exitcode "${OPTIONAL_ARGS[@]}"
```

File: .github/workflows/terraform.yml (hunk 1 above)
Comment (chbul) [reply .github/workflows/terraform.yml:43]: > I agree 👍
> 
> How about something like this?
> 
> ```
> The path, relative to the working directory, of a variable definitions (`.tfvars`) file containing values for variables declared in the root module.
> ```

If you agree, I'd like to move "(`.tfvars`)" after "file" so it doesn't split the sentence

Comment (hknutsen) [reply .github/workflows/terraform.yml]: My intention was that you could read it as either "variable definitions file" or "`.tfvars` file".

Also, I took it word-for-word from [this heading](https://developer.hashicorp.com/terraform/language/values/variables#variable-definitions-tfvars-files) in the official docs.

**Review by sefornes — APPROVED** [review]

**Review by sefornes — APPROVED** [review]

**Review by hknutsen — APPROVED** [review]
